### DISCUSSION TITLE
Allow non-contributors to add labels to issues/pull requests

### DISCUSSION THREAD

**Discussion by RixInGithub** [community/community#125382] [Projects and Issues] [answered]
### Select Topic Area

Question

### Body

So, I have a GitHub repo that I think people may want to interact with. And I want those people which aren't contributors, to add *some* labels. Maybe like a separate setting for each label?

**Comment by spenserblack** [comment, marked as answer]
[Issue templates/forms](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository) allow you to assign initial labels for various issue types. I think this is enough to handle the majority of cases, because I think generally you only want non-maintainers to assign labels at issue *creation,* not give them the ability to change labels after creation.

If you to grant users the ability to assign labels *after* the issue is created, you can write a GitHub Workflow that listens to issue comments getting created and checks for text like `/label bug`. I could write out an example workflow if that's the way you want to go, but I'd recommend writing issue templates or issue forms instead.

---

As far as pull requests go you might want to set up an auto-labeler. [Here's an action to do that](https://github.com/actions/labeler), or you could write your own script. Personally, I wrote an auto-labeling workflow that compares an old and new file and adds the appropriate label if the changes look good.

**Reply by RixInGithub** [reply]
Good enough, I suppose

**Comment by roke-julian-lockhart** [comment]
> [`#discussion-6718052`](https://github.com/orgs/community/discussions/125382#discussion-6718052)

@RixInGithub, see [`stackoverflow.com/q/13829466`](https://stackoverflow.com/q/13829466/9731176).
